sys/libjade/build.rs

```diff
@@ -1,211 +1,184 @@
+            "x86_64-apple-darwin" => svec!["-target", "x86_64-apple-darwin"],
+            _ => panic!("Unsupported cross compilation target {s}"),
+        })
+        .unwrap_or_default();
+
+    let files = svec![
+        "sha256.s",
+        "x25519_ref.s",
+        "x25519_mulx.s",
+        "sha3_224_ref.s",
+        "sha3_256_ref.s",
+        "sha3_384_ref.s",
+        "sha3_512_ref.s",
+        "chacha20_ref.s",
+        "poly1305_ref.s",
+        "kyber_kyber768_ref.s",
+    ];
+    compile_files("libjade.a", &files, out_path, &args);
+
+    if platform.simd256 {
+        let files256 = svec![
+            "sha3_224_avx2.s",
+            "sha3_256_avx2.s",
+            "sha3_384_avx2.s",
+            "sha3_512_avx2.s",
+            "chacha20_avx2.s",
+            "poly1305_avx2.s",
+        ];
+
+        let mut simd256_flags = args.clone();
+        append_simd256_flags(&mut simd256_flags);
+        compile_files("libjade_256.a", &files256, out_path, &simd256_flags);
     }
 
-    pub fn simd256_support() -> bool {
-        std::arch::is_x86_feature_detected!("avx2")
+    if platform.simd128 {
+        let files128 = svec!["chacha20_avx.s", "poly1305_avx.s",];
+
+        let mut simd128_flags = args.clone();
+        append_simd128_flags(&mut simd128_flags);
+        compile_files("libjade_128.a", &files128, out_path, &simd128_flags);
     }
 }
 
-#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-pub fn main() {
-    use std::{env, path::Path};
-    use x64_build::*;
+#[derive(Debug, Clone, Copy)]
+struct Platform {
+    simd128: bool,
+    simd256: bool,
+}
 
+pub fn main() -> Result<(), u8> {
     // Get ENV variables
     let home_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
     let home_path = Path::new(&home_dir);
     let out_dir = env::var("OUT_DIR").unwrap();
     let out_path = Path::new(&out_dir);
     let target = env::var("TARGET").unwrap();
+    let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
     let host = env::var("HOST").unwrap();
 
-    let cross_target = if target != host { Some(target) } else { None };
-    if cross_target.is_none() && host == "aarch64-apple-darwin" {
-        panic!("libjade does not support aarch64-apple-darwin");
+    if target_arch != "x86_64" && target_arch != "x86" {
+        eprintln!(" ! Only x86 and x64 CPUs are supported !");
+        return Err(1);
     }
 
-    if simd128_support() {
-        println!("cargo:rustc-cfg=simd128");
-    }
-    if simd256_support() {
-        println!("cargo:rustc-cfg=simd256");
-    }
+    let cross_target = if target != host {
+        Some(target.clone())
+    } else {
+        None
+    };
+
+    // If cross compiling, we assume to have it all.
+    let platform = if cross_target.is_some() {
+        Platform {
+            simd128: true,
+            simd256: true,
+        }
+    } else {
+        Platform {
+            simd128: std::arch::is_x86_feature_detected!("avx"),
```

Comment:
This doesn't work.
The build script is built for the host, not the target platform. So this throws an error when cross compiling.
You need to guard this with the host platform.

Reply:
Replaced with functions from platform package, should be fixed now.